### ISSUE TITLE
Record history of hparams metrics

### ISSUE THREAD

**Issue by rohaldbUni** [lanpa/tensorboardX#475]
I'm unsure how to record metrics during training such that when I run `add_hparams()` at the end, the `show metrics` graphs in the HPARAMS tab contains more than a single value.  What I would like to do is something like this:
```
w.add_scalar("loss", 10, 0)
w.add_scalar("loss", 11, 1)
w.add_scalar("loss", 12, 2)
w.add_hparams({'lr': X, 'bsize': Y, 'n_hidden': Z}, {'loss': 13})
```
And then the `show metrics` graph would have the four points [10,11,12,13] plotted for the loss metric. 

Thanks!

**Comment by lanpa** [comment]
Because `loss` used in the dictionary and `loss` in add_scalar have the same name.
You can fix that by adding a prefix like 
`w.add_hparams({'lr': X, 'bsize': Y, 'n_hidden': Z}, {'hparam/loss': 13})`

**Comment by rohaldbUni** [comment]
Sorry, that didn't work. In the above PR, you added 

> the key used here should be unique .... otherwise the value you added by `add_scalar` will be displayed in hparam plugin

But if I do this where the key is the same

```
w.add_hparams({'lr': X, 'bsize': Y, 'n_hidden': Z}, {'loss': 13})
w.add_scalar("loss", 14, 0)
```

Then loss value in the hparams tab is 13, not 14. Your suggestion of prepending the key with `hparam/` also didn't work (and intuitively if they are based off unique keys then I assume it shouldn't). 

```
w.add_scalar("loss", 10, 0)
w.add_scalar("loss", 11, 1)
w.add_scalar("loss", 12, 2)
w.add_hparams({'lr': X, 'bsize': Y, 'n_hidden': Z}, {'hparam/loss': 13})
```

Yields only a single value in the hparams tab of 13. 

Thanks for your help by the way!

**Comment by lanpa** [comment]
"Recall my goal is to have the hparams show metric display all four values, [10,11,12,13]. In the above PR, you added"

I'm not sure is that doable in official TensorBoard implementation. I think the reasonable way is to represent a metric with a single value. I can try to work on your need if you found some example in tensorflow first.

**Comment by asford** [comment]
@rohaldbUni Then TensorBoard HParams plugin will aggregate and report scalars from your run into the reporting page, you just need to manually write the experimental overview into the tensorboard event stream. For your example:

```python

 exp, ssi, sei = tensorboardX.summary.hparams(
{'lr': X, 'bsize': Y, 'n_hidden': Z}, {'loss': 13})
            _flat(config),
            {"loss": None, "evaluate/accuracy": None, "train/loss": None},
        )

        summary_writer.file_writer.add_summary(exp)
        summary_writer.file_writer.add_summary(ssi)

**Comment by asford** [comment]
@lanpa Is this repo the primary development point for tensorboardX, or is the module being folded into pytorch mainline development? I've noticed a bit of discussion there on https://github.com/pytorch/pytorch/pull/23134 and https://github.com/pytorch/pytorch/issues/16838. 

I'm currently using a method like the one above for hparam reporting in pytorch, but I'd be happy to expand and document the current tensorboardX interface to cover this use case.
